ERCS/erc-7651.md

```diff
@@ -0,0 +1,349 @@
+  /// @param amountOrId_ The fractional value or a distinct NFT id to transfer
+  /// @return True if the operation was successful
+  function transferFrom(
+    address from_,
+    address to_,
+    uint256 amountOrId_
+  ) external returns (bool success);
+
+  /// @notice Transfer fractional tokens from one address to another
+  /// @dev The operation should revert if amount is less than the balance of `from_`
+  /// @param to_ The address to transfer fractional tokens to
+  /// @param amount_ The fractional value to transfer
+  /// @return True if the operation was successful
+  function transfer(address to_, uint256 amount_) external returns (bool success);
+
+  /// @notice Transfers the ownership of an NFT from one address to another address
+  /// @dev Throws unless `msg.sender` is the current owner, an authorized
+  ///      operator, or the approved address for this NFT
+  /// @dev Throws if `from_` is not the current owner
+  /// @dev Throws if `to_` is the zero address
+  /// @dev Throws if `tokenId_` is not a valid NFT
+  /// @dev When transfer is complete, this function checks if `to_` is a
+  ///      smart contract (code size > 0). If so, it calls `onERC721Received`
+  ///      on `to_` and throws if the return value is not
+  ///      `bytes4(keccak256("onERC721Received(address,uint256,bytes)"))`.
+  /// @param from_ The address to transfer the NFT from
+  /// @param to_ The address to transfer the NFT to
+  /// @param tokenId_ The NFT to transfer
+  /// @param data_ Additional data with no specified format, sent in call to `to_`
+  function safeTransferFrom(
+    address from_,
+    address to_,
+    uint256 id_,
+    bytes calldata data_
+  ) external;
+
+  /// @notice Transfers the ownership of an NFT from one address to another address
+  /// @dev This is identical to the above function safeTransferFrom interface
+  ///      though must pass empty bytes as data to `to_`
+  /// @param from_ The address to transfer the NFT from
+  /// @param to_ The address to transfer the NFT to
+  /// @param tokenId_ The NFT to transfer
+  function safeTransferFrom(address from_, address to_, uint256 id_) external;
+}
+
+interface IERC165 {
+    /// @notice Query if a contract implements an interface
+    /// @param interfaceID_ The interface identifier, as specified in ERC-165
+    /// @dev Interface identification is specified in ERC-165. This function
+    ///      uses less than 30,000 gas.
+    /// @return `true` if the contract implements `interfaceID` and
+    ///         `interfaceID` is not 0xffffffff, `false` otherwise
+    function supportsInterface(bytes4 interfaceID_) external view returns (bool);
+}
+```
+
+### Fractionally Represented Non-Fungible Token Metadata Interface
+
+This is a RECOMMENDED interface, identical in definition to the [ERC-721](./eip-721.md) Metadata Interface. Rather than using this interface directly, a distinct metadata interface should be used here to avoid confusion surrounding ERC-721 inheritance.
+
```

Comment:
I think we shall mention the ERC165 interfaceId will therefore be the same

Reply:
A solution used in other cases is to enforce the interfaceId using a magic number, but I think that it would introduce more issues. I suggest that the author adds an extra function that does nothing but is there only to produce a different interfaceId, like
```
function isIERC7651Metadata() external pure returns(bool);
```
that can only return `true`.

Reply:
I decided to just add clarification for now. Given this is a draft, I figure it would be better to discuss the possibility of introducing some means of distinguishing the interfaceId's on the discussion thread.